lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -1,436 +1,348 @@
-                              ex(?<on>on)?(?<s>s)?\s(?<exons>[0-9]+(?<dgs>\sto\s[0-9]+)?)\s
-                              (?<variant>del|dup|ins)/ix.freeze
-
-          DEPRECATED_BRCA_NAMES_REGEX = /B1|BR1|BRCA\s1|B2|BR2|BRCA\s2/i.freeze
-
-          DELIMETER_REGEX = /[&\n+,;]|and|IFD/i.freeze
+          include Import::Helpers::Brca::Providers::Rj7::Constants
 
           def process_fields(record)
             genotype = Import::Brca::Core::GenotypeBrca.new(record)
-            genotype.add_passthrough_fields(record.mapped_fields,
-                                            record.raw_fields,
-                                            PASS_THROUGH_FIELDS)
-            add_organisationcode_testresult(genotype)
-            add_moleculartestingtype(genotype, record)
-            process_genetictestcope(genotype, record)
-            res = process_variants_from_record(genotype, record)
-            res.each { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
-          end
-
-          def add_organisationcode_testresult(genotype)
-            genotype.attribute_map['organisationcode_testresult'] = '697N0'
-          end
+            # records using new importer should only have SRIs starting with V
+            return unless record.raw_fields['servicereportidentifier'].start_with?('V')
 
-          def add_moleculartestingtype(genotype, record)
-            return if record.raw_fields['moleculartestingtype'].nil?
+            genotype.add_passthrough_fields(record.mapped_fields, record.raw_fields, PASS_THROUGH_FIELDS)
+            assign_test_type(genotype, record)
+            genotype = assign_test_scope(genotype, record)
 
-            moltesttype = record.raw_fields['moleculartestingtype']
-            if moltesttype.scan(/unaf|pred/i).size.positive?
-              genotype.add_molecular_testing_type_strict(:predictive)
-            elsif moltesttype.scan(/affected|conf/i).size.positive?
-              genotype.add_molecular_testing_type_strict(:diagnostic)
-            end
-          end
+            genotypes = fill_genotypes(genotype, record)
 
-          def process_genetictestcope(genotype, record)
-            if ashkenazi?(record)
-              genotype.add_test_scope(:aj_screen)
-            elsif polish?(record)
-              genotype.add_test_scope(:polish_screen)
-            elsif targeted_test?(record)
-              genotype.add_test_scope(:targeted_mutation)
-            elsif full_screen?(record)
-              genotype.add_test_scope(:full_screen)
-            elsif void_genetictestscope?(record)
-              @logger.debug 'Unknown moleculartestingtype'
-              genotype.add_test_scope(:no_genetictestscope)
+            genotypes.each do |single_genotype|
+              process_variants(single_genotype, record)
+              @persister.integrate_and_store(single_genotype)
             end
+            genotypes
```

Comment:
Why would you need this @lauramccluskey1 ?

Reply:
I needed it to be able to test the unittests for that function